homeassistant/components/abode/camera.py

```diff
@@ -7,19 +7,25 @@
 from homeassistant.components.camera import Camera
 from homeassistant.util import Throttle
 
-from . import DOMAIN as ABODE_DOMAIN, AbodeDevice
+from . import AbodeDevice
+from .const import DOMAIN
 
 MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=90)
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up Abode camera devices."""
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up a camera for an Abode device."""
     import abodepy.helpers.constants as CONST
     import abodepy.helpers.timeline as TIMELINE
 
-    data = hass.data[ABODE_DOMAIN]
+    data = hass.data[DOMAIN]
 
     devices = []
     for device in data.abode.get_devices(generic_type=CONST.TYPE_CAMERA):
```

Comment:
I/O?

Reply:
I don't believe this one does since a `generic_type` is passed.

https://github.com/MisterWil/abodepy/blob/6f84bb428fd1da98855f55083cd427bebbcc57ae/abodepy/__init__.py#L241

Reply:
Do we know that devices are already cached?
https://github.com/MisterWil/abodepy/blob/6f84bb428fd1da98855f55083cd427bebbcc57ae/abodepy/__init__.py#L192

Reply:
I believe so because when `abodepy.Abode` is [called](https://github.com/home-assistant/home-assistant/blob/e005f6f23a3ff9ff052afd317891da206ea618e0/homeassistant/components/abode/__init__.py#L107) during the setup of `AbodeSystem`, `get_devices` is set to `True` which should populate `_devices`.